src/custom/components/Tokens/FiatBalanceCell.tsx

```diff
@@ -0,0 +1,41 @@
+import { Trans } from '@lingui/macro'
+import { Token, CurrencyAmount } from '@uniswap/sdk-core'
+import { useHigherUSDValue } from 'hooks/useUSDCPrice'
+import { FIAT_PRECISION } from 'constants/index'
+import { useActiveWeb3React } from 'hooks/web3'
+import { BalanceValue, InfoCircle, FiatValue } from './styled'
+import { formatMax, formatSmart } from 'utils/format'
+import { MouseoverTooltip } from 'components/Tooltip'
+import Loader from 'components/Loader'
+
+type FiatBalanceCellProps = {
+  balance: CurrencyAmount<Token> | undefined
+}
+
+export default function FiatBalanceCell({ balance }: FiatBalanceCellProps) {
+  const { account } = useActiveWeb3React()
+  const hasBalance = balance?.greaterThan(0)
+  const fiatValue = useHigherUSDValue(balance)
+  const formattedFiatValue = formatSmart(fiatValue, FIAT_PRECISION, {
+    thousandSeparator: true,
+    isLocaleAware: true,
+  })
+
+  if (!balance && account) return <Loader />
```

Comment:
The thing is that those 2 describe different situations. If balance is undefined it means its not loaded yet and we need to detect that, and hasBalance also detects if the balance is loaded and is 0 which we also need. They do overlap in case where the balance is not loaded yet (both undefined) but its fine since then it will return on the first condition